### PR TITLE
Improves assert_delivered_email and adds assert_delivered_with

### LINKED ISSUE #74: Fix options with assert_delivered_email
It should normalize the email addresses, and should only match against the keys that are in the options. For asserting just what you pass in, do something like this

``` elixir
defmacro assert_delivered_email(email_params) do
    # Normalize the address as well
    quote do
      import ExUnit.Assertions
      assert_received({:delivered_email, unquote(email_params)})
    end
  end
```

### PR DESCRIPTION
**Pull request by jbernardo95**
Resolves #74.

### CODE REVIEW

**Comment by optikfluffel** [comment]
Any updates on that one? 😸

**Comment by jbernardo95** [comment]
@paulcsmith ^

**Comment by jbernardo95** [comment]
@optikfluffel I guess this is related with #233, am I right ?

File: lib/bamboo/test.ex [hunk 1]
```diff
@@ -146,7 +144,7 @@ defmodule Bamboo.Test do
   Checks whether an email was delivered.
 
   Must be used with the `Bamboo.TestAdapter` or this will never pass. In case you
-  are delivering from another process, the assertion waits up to 100ms before
+  are delivering from another process, the assertion waits up to 150ms before
```
Comment (paulcsmith) [inline lib/bamboo/test.ex]: Was there a reason for changing this to 150ms?

File: test/lib/bamboo/adapters/test_adapter_test.exs [hunk 2]
```diff
@@ -106,6 +114,33 @@ defmodule Bamboo.TestAdapterTest do
...
   end
 
+  test "assert_delivered_with with no delivered emails" do
+    try do
+      assert_delivered_with from: {nil, "foo@bar.com"}
+    rescue
+      error in [ExUnit.AssertionError] ->
+        assert error.message =~ "0 emails delivered"
+    else
+      _ -> flunk "assert_delivered_email should failed"
```
Comment (paulcsmith) [inline test/lib/bamboo/adapters/test_adapter_test.exs]: `should have failed`? both here and in the next test

**Comment by paulcsmith** [comment]
@jbernardo95 Thanks for tackling his. Just a few small comments and then I'll merge this in!

Also,  I think that feature is somewhat related, but I would probably test token generation in a unit test. It's easier to isolate. Then you could use this new function you wrote to just test the subject and skip the body in integration tests

File: lib/bamboo/test.ex (hunk 1 above)
Comment (jbernardo95) [inline lib/bamboo/test.ex]: At the time I ran the tests, 100ms seemed to not be enough. But I ran the tests now 3 times in a row with the 100ms and they seem fine.

Changed back to 100ms.

File: lib/bamboo/test.ex [hunk 3]
```diff
@@ -159,33 +157,46 @@ defmodule Bamboo.Test do
...
+    end
   end
 
-  defp do_assert_delivered_email(email) do
-    receive do
-      {:delivered_email, ^email} -> true
-    after
-      @timeout -> flunk_with_email_list(email)
+  @doc """
+  Checks whether an email its params equal to the ones provided.
```
Comment (paulcsmith) [inline lib/bamboo/test.ex]: One more small typo: Check whether an email's params are equal to the ones provided

**Comment by jbernardo95** [comment]
@paulcsmith Updated !
CircleCI seems to be stuck...

**Comment by jbernardo95** [comment]
@paulcsmith Done, let's see if this helps.

**Comment by jbernardo95** [comment]
@paulcsmith It didn't help, this seems to be a CircleCI <> GitHub issue.

The other new PR I submitted isn't getting any status either. Checked back at [CircleCI](https://circleci.com/gh/thoughtbot/bamboo) and it seems to not be getting GitHub notifications of new code.

**Comment by paulcsmith** [comment]
@jbernardo95 I'll have to figure this out some other time. It appears to be working fine though so I'll merge this in. Thank you so much!

**Pull request merged**